### ISSUE TITLE
Stateful RNNs and Backpropagation through time

### ISSUE THREAD

**Issue by leftys** [keras-team/keras#8945]
Hello,

I have quite common use case for a RNN/LSTM network: 'infinite' time serie, with 'label' on each time step. The future data are dependent mostly on recent time steps, but also on very old data. I would like to train a neural network to predict the time serie, but with Keras, I can only use stateless RNN with very limited timestep count, or stateful RNN, that can't backpropagate through time and therefore 'learn what to remember'. 

It would be good to implement some sort of perhaps truncated back propagation through time for stateful RNNs. Without that, LSTMs, which are good for long term dependencies, are not that useful.

What do you think about it?
